website/docs/en/config/devtool.mdx

```diff
@@ -4,54 +4,93 @@ import WebpackLicense from '@components/WebpackLicense';
+- **Required** → Proceed to [Step 2](#step-2-define-debugging-requirements)
+
+### Step 2: Define Debugging Requirements
+
+- **Module-level positioning only** → Set `devtool: 'eval'`
+  - Each module executed via `eval()` with `//# sourceURL` comment
+  - Extremely fast build speed
+- **Full source code mapping needed** → Proceed to [Step 3](#step-3-configure-sourcemap)
+
+### Step 3: Configure SourceMap
+
+Set `devtool: 'source-map'`, A full SourceMap is emitted as a separate file. It adds a `//# sourceMapURL` comment to the bundle so development tools know where to find it.
+
+It also supports combination with the following modifiers to improve performance and control SourceMap generation.
+
+Performance optimization modifiers, to speed up the build, usually used in development environments:
+
+| Modifier | Effect                                                                                                                                       | Performance improvement |
+| -------- | -------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------- |
+| eval     | Each module is executed with `eval()` and a SourceMap is added as a DataUrl to the `eval()`, avoiding chunk-level multiple SourceMap concate | ⚡⚡⚡                  |
+| cheap    | Maps line numbers only (no columns), ignores source maps from loaders                                                                        | ⚡⚡                    |
+| module   | Processes source maps from loaders to map to original code (line-only mapping)                                                               | ⚡                      |
+
+Functional modifiers, to control SourceMap generation, usually used in production environments:
+
+| Modifier  | Effect                                                                                                                                      |
+| --------- | ------------------------------------------------------------------------------------------------------------------------------------------- |
+| hidden    | SourceMap is emitted as a separate file, but no `//# sourceMappingURL=[url]` comment is added to the bundle, protecting source code privacy |
+| inline    | SourceMap is added as a DataUrl to the bundle                                                                                               |
+| nosources | SourceMap is created without the `sourcesContent` in it                                                                                     |
+| debugids  | SourceMap is created with the `debugId` in it                                                                                               |
+
+We expect a certain pattern when validate devtool name, pay attention and don't mix up the sequence of devtool string. The pattern is: `[inline-|hidden-|eval-][nosources-][cheap-[module-]]source-map[-debugids]`.
```

Comment:
`[inline-|hidden-|eval-][nosources-][cheap-[module-]]source-map[-debugids]` is too complex for most users